force-app/main/default/lwc/tableauViz/tableauViz.js

```diff
@@ -8,86 +8,191 @@ import templateMain from './tableauViz.html';
-    @api sfAdvancedFilter;
-
-    viz;
-    advancedFilterValue;
+    //#endregion Special LWC variable
+
+    //#region Filter variables
+    // Need to maintain an independent list that wire service can react to changes on
+    _recordFilterFields = [];
+    _recordFilters = new Map();
+    _recordFiltersResolved = true; // If there are no fields, then they are resolved
+    // this is a bit of a duplication of the above but we do this for manageability of the 'truth'
+    _filters = new Map();
+
+    // We need these cause the property editor UI is not expressive enough yet
+    // Adding another filter involves two properties here, two api getter/setter properties, and an additional line in validateInputs
+    _sfAdvancedFilter;
+    _tabAdvancedFilter;
+    //#endregion Filter variables
+
+    // Used in the error html page
     errorMessage;
-    isLibLoaded = false;
 
-    // Use hashnames for private fields when that is accepted
-    _showTabs = false;
-    _showToolbar = false;
-    _filterOnRecordId = false;
+    _viz;
+    _isLibLoaded = false;
+    //#endregion Member Variables
 
     @wire(getRecord, {
         recordId: '$recordId',
-        fields: '$sfAdvancedFilter'
+        fields: '$_recordFilterFields'
     })
     getRecord({ error, data }) {
         if (data) {
-            this.advancedFilterValue = getFieldValue(
-                data,
-                this.sfAdvancedFilter
-            );
-            if (this.advancedFilterValue === undefined) {
-                this.errorMessage = `Failed to retrieve value for field ${this.sfAdvancedFilter}`;
-            } else {
-                this.renderViz();
-            }
+            // iterate over each recordFilterField and get the value
+            this._recordFilters.forEach((tableauField, recordField) => {
+                const fieldValue = getFieldValue(data, recordField);
+                if (fieldValue === undefined) {
+                    this.errorMessage = `Failed to retrieve value for field ${recordField}`;
+                } else {
+                    this.addFilter(tableauField, fieldValue);
+                }
+            });
+            this._recordFiltersResolved = true;
+            this.renderViz();
         } else if (error) {
             this.errorMessage = `Failed to retrieve record data: ${reduceErrors(
                 error
             )}`;
         }
     }
 
-    // In JavaScript, there are six falsy values:
-    // false, 0, '', null, undefined, and NaN. Everything else is truthy.
-    // LWC can sometimes return 'false' as a string and we need to treat it as false
-    // the !! operator converts any object to Boolean type
-    static booleanNormalize(val) {
-        if (typeof val == 'string' && val.toLowerCase() === 'false') {
-            return false;
+    //#region Getter & Setter Helpers
+    // https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes
+    _setBooleanAttribute(name, v) {
```

Comment:
I am confused here. This is an explicit recommended pattern? https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.js_props_html_attributes. What are you saying I am doing that is "wrong"

Reply:
As a general rule, we should avoid manipulating the DOM manually whenever possible because the framework is in charge of that.

The original exception to this rule was third party integrations with the use of `lwc:dom="manual"` (see "Manipulate the DOM with JavaScript" section from [this page](https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.js_third_party_library)).

The page that you shared is another exception to this rule for the sake of accessibility:
> Allowing properties to appear as attributes is especially important when creating accessible components, because screen readers and other assistive technologies use HTML attributes.

I'll admit that the documentation isn't super clear about the accessibility use case (I'll give that feedback to the docs team) but I don't think that we can justify the use of this technique for the Tableau component. We don't need to surface those attributes in the DOM because the user doesn't need them and we can test with other means (behaviour tests).

Reply:
Eventually, I would see being able to use the viz by embedding like:
<tableau-viz id="jurisdictionViz" 
      url='https://us-west-2a.online.tableau.com/t/alpodev/views/Regional/College'
      tabs="hidden" toolbar="bottom" debug oncompleted="handleCompletedEvent" onMarksSelected="marksSelected" tabs="hidden" language='en' display-count="display-count" origin='viz_share_link'>
    </tableau-viz>

(ignore a couple of details ... that is from a prototype outside of LWC). It seems like having these attributes in sync is a good idea. Not sure how to measure "justify". I definitely didn't do it for test purposes but more for usage.

Reply:
`<tableau-viz id="jurisdictionViz" 
      url='https://us-west-2a.online.tableau.com/t/alpodev/views/Regional/College'
      tabs="hidden" toolbar="bottom" debug oncompleted="handleCompletedEvent" onMarksSelected="marksSelected" tabs="hidden" language='en' display-count="display-count" origin='viz_share_link'>
    </tableau-viz>`

Reply:
Oh ok, this makes total sense. Thanks for explaining.

For some reason, I thought you only did that for testing when I read this comment on `_checkBooleanAttribute(name)` and the tests:
```js
// hasAttribute is not available on the LightningElement from testing so ... workaround
```

I think that it's fine then, let's keep this.